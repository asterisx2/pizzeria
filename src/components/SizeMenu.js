import React from 'react';
import PizzSizeChooser from './PizzaSizeChooser';

export default ({ 
  pizzaSizes,
  handleSelectSize, 
  selectedSize 
}) => {
  return (
    <section>
      <div 
        className="d-flex flex-column">
        <h4 
          className="border-bottom-black text-left pb-2 mb-2">
          Select a size
        </h4>
        {pizzaSizes.map(pizzaSize => {
          return (
            <PizzSizeChooser
              key={pizzaSize.id}
              pizzaId={pizzaSize.id}
              sizeName={pizzaSize.sizeName}
              pizzaImg={pizzaSize.imgPath}
              checkBoxName={'pizzaSize'}
              basePrice={pizzaSize.basePrice}
              selected={selectedSize === pizzaSize.sizeName}
              onSelected={handleSelectSize}
            />
          );
        })}
      </div>
    </section>
  );
};
