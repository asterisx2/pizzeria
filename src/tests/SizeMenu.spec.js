import React from 'react';
import SizeMenu from '../components/SizeMenu';
import renderer from 'react-test-renderer';

describe('SizeMenu Component', () => {
  let pizzaSizes = [
    {
      id: 1,
      sizeName: 'small',
      basePrice: 1.00,
      imgPath: '../../pizza_round.png'
    },
    {
      id: 2,
      sizeName: 'medium',
      basePrice: 1.50,
      imgPath: '../../pizza_round.png'
    },
    {
      id: 3,
      sizeName: 'large',
      basePrice: 2.00,
      imgPath: '../../pizza_round.png'
    }
  ];

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <SizeMenu
          pizzaSizes={pizzaSizes}
          handleSelectSize={() => { }}
          selectedSize={'small'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO
  it('handles size selection correctly', () => { });
});
