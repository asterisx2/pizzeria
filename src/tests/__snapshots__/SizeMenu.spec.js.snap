// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeMenu Component renders without crashing 1`] = `
<section>
  <div
    className="d-flex flex-column"
  >
    <h4
      className="border-bottom-black text-left pb-2 mb-2"
    >
      Select a size
    </h4>
    <div
      className="border d-flex justify-content-between align-items-center mt-2 p-2 hover-dark selected"
      onClick={[Function]}
    >
      <img
        alt=""
        height="60px"
        src="../../pizza_round.png"
      />
      <div
        className="d-flex flex-column text-right"
      >
        <span
          className="ml-2 text-secondary font-large pl-6 pr-2"
        >
          small
        </span>
        <span
          className="text-primary font-italic font-weight-light"
        >
          Starting from €
          1.00
        </span>
      </div>
    </div>
    <div
      className="border d-flex justify-content-between align-items-center mt-2 p-2 hover-dark"
      onClick={[Function]}
    >
      <img
        alt=""
        height="70px"
        src="../../pizza_round.png"
      />
      <div
        className="d-flex flex-column text-right"
      >
        <span
          className="ml-2 text-secondary font-large pl-6 pr-2"
        >
          medium
        </span>
        <span
          className="text-primary font-italic font-weight-light"
        >
          Starting from €
          1.50
        </span>
      </div>
    </div>
    <div
      className="border d-flex justify-content-between align-items-center mt-2 p-2 hover-dark"
      onClick={[Function]}
    >
      <img
        alt=""
        height="80px"
        src="../../pizza_round.png"
      />
      <div
        className="d-flex flex-column text-right"
      >
        <span
          className="ml-2 text-secondary font-large pl-6 pr-2"
        >
          large
        </span>
        <span
          className="text-primary font-italic font-weight-light"
        >
          Starting from €
          2.00
        </span>
      </div>
    </div>
  </div>
</section>
`;
